full-service/src/service/transaction.rs

```diff
@@ -468,6 +468,22 @@ where
         let conn = pooled_conn.deref_mut();
 
         exclusive_transaction(conn, |conn| {
+            if Account::get(&AccountID(account_id_hex.to_string()), conn)?
```

Comment:
📌  This is the main logic enforcing the match between `requires_spend_subaddresses` on an account, and providing `spend_from_subaddress` to the `build` methods